clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/view/DcosJobProvider.groovy

```diff
@@ -0,0 +1,80 @@
+package com.netflix.spinnaker.clouddriver.dcos.provider.view
+
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.netflix.spinnaker.clouddriver.dcos.DcosClientProvider
+import com.netflix.spinnaker.clouddriver.dcos.DcosCloudProvider
+import com.netflix.spinnaker.clouddriver.dcos.model.DcosJobStatus
+import com.netflix.spinnaker.clouddriver.model.JobProvider
+import mesosphere.dcos.client.DCOSException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class DcosJobProvider implements JobProvider<DcosJobStatus> {
+  static final JOB_FRAMEWORK = "metronome"
+
+  final String platform = DcosCloudProvider.ID
+
+  private final DcosClientProvider dcosClientProvider
+
+  @Autowired
+  DcosJobProvider(DcosClientProvider dcosClientProvider) {
+    this.dcosClientProvider = dcosClientProvider
+  }
+
+  @Override
+  DcosJobStatus collectJob(String account, String location, String id) {
+    def dcosClient = dcosClientProvider.getDcosClient(account)
+    def jobResponse = dcosClient.getJob(location, ['activeRuns','history'])
+
+    return new DcosJobStatus(jobResponse, id, account)
+  }
+
+  @Override
+  Map<String, Object> getFileContents(String account, String location, String id, String fileName) {
+    def dcosClient = dcosClientProvider.getDcosClient(account)
+    def taskName = "${id}.${location}".toString()
+    def masterState = dcosClient.getMasterState()
+
+    def metronomeFramework = masterState.getFrameworks().find {
+      framework -> (JOB_FRAMEWORK == framework.getName())
+    }
+
+    def jobTask = metronomeFramework.getCompleted_tasks().find {
+      task -> (taskName == task.getName())
+    }
+
+    def agentState = dcosClient.getAgentState(jobTask.getSlave_id())
+
+    metronomeFramework = agentState.getCompleted_frameworks().find {
```

Comment:
Nah it's good, just didn't understand the differences. Thanks for the explanation.